meerk40t/tools/livinghinges.py

```diff
@@ -0,0 +1,1129 @@
+                        # Adjust x - value
+                        if dy != 0:
+                            new_cx_clipped = new_cx + dx / dy * (new_cy_clipped - new_cy)
+                            new_ex_clipped = new_ex + dx / dy * (new_ey_clipped - new_ey)
+
+                        new_cx = new_cx_clipped
+                        new_cy = new_cy_clipped
+                        new_ex = new_ex_clipped
+                        new_ey = new_ey_clipped
+                        if min(new_cy, new_ey) == ymax and dy != 0:
+                            # Outward...
+                            pass
+                        elif max(new_cy, new_ey) == ymin and dy != 0:
+                            # Outward...
+                            pass
+                        else:
+                            if new_cx != current_x or new_cy != current_y:
+                                # Needs a move
+                                add_move(newpath, Point(new_cx, new_cy))
+                            newpath.line(Point(new_ex, new_ey))
+                    if current_x != e.end[0] or current_y != e.end[1]:
+                        add_move(newpath, e.end)
+                    partial_deleted += 1
+                current_x = e.end[0]
+                current_y = e.end[1]
+            elif isinstance(e, Close):
+                newpath.closed()
+                not_deleted += 1
+            elif isinstance(e, QuadraticBezier):
+                statex, statey = outside(segbb, clipbb)
+                if statex == "outside" and statey == "outside":
+                    # Fully outside, so drop
+                    if current_x != e.end[0] or current_y != e.end[1]:
+                        add_move(newpath, e.end)
+                    fully_deleted += 1
+                elif statex == "inside" and statey == "inside":
+                    # Fully inside, so append
+                    newpath.quad(e.control, e.end)
+                    not_deleted += 1
+                else:
+                    approximate_line(e, current_x, current_y)
+                current_x = e.end[0]
+                current_y = e.end[1]
+            elif isinstance(e, CubicBezier):
+                statex, statey = outside(segbb, clipbb)
+                if statex == "outside" and statey == "outside":
+                    # Fully outside, so drop
+                    if current_x != e.end[0] or current_y != e.end[1]:
+                        add_move(newpath, e.end)
+                    fully_deleted += 1
+                elif statex == "inside" and statey == "inside":
+                    # Fully inside, so append
+                    newpath.cubic(e.control1, e.control2, e.end)
+                    not_deleted += 1
+                else:
+                    approximate_line(e, current_x, current_y)
+                    partial_deleted += 1
+                current_x = e.end[0]
+                current_y = e.end[1]
+            elif isinstance(e, Arc):
```

Comment:
There's also a command that replaces arcs with the cubic curves before hand since that's generally an easier way to avoid them.